fortran/src/H5Pf.c

```diff
@@ -4669,16 +4669,16 @@ h5pget_fapl_mpio_c(hid_t_f *prp_id, int_f *comm, int_f *info)
  * SOURCE
  */
 int_f
-h5pset_mpi_params_c(hid_t_f *prp_id, int_f *comm, int_f *info)
+h5pset_mpi_params_c(hid_t_f *prp_id, void *comm, void *info)
 /******/
 {
     int      ret_value = -1;
     hid_t    c_prp_id;
     herr_t   ret;
     MPI_Comm c_comm;
     MPI_Info c_info;
-    c_comm = MPI_Comm_f2c(*comm);
-    c_info = MPI_Info_f2c(*info);
+    c_comm = MPI_Comm_f2c(*((int *)comm));
```

Comment:
Not at all relevant to this PR, but @brtnfld it seems problematic to always assume here that `int` converts to `MPI_Fint`